python/lsst/afw/table/testUtils.py

```diff
@@ -0,0 +1,88 @@
+#
+# LSST Data Management System
+# Copyright 2016 AURA/LSST.
+#
+# This product includes software developed by the
+# LSST Project (http://www.lsst.org/).
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the LSST License Statement and
+# the GNU General Public License along with this program.  If not,
+# see <https://www.lsstcorp.org/LegalNotices/>.
+#
+from __future__ import absolute_import, division, print_function
+
+import lsst.utils.tests
+import difflib
+from .tableLib import Schema
+
+__all__ = ["assertSchemasEqual", "diffSchemas"]
+
+def diffSchemas(schema1, schema2, flags=Schema.IDENTICAL):
+    """Return a string diff of two schemas.
+
+    Parameters
+    ----------
+    schema1 : :py:class:`lsst.afw.table.Schema`
+        First schema to diff.  Items appearing only in this schema
+        will be prefixed with "-" in the diff.
+    schema2 : :py:class:`lsst.afw.table.Schema`
+        Second schema to diff.  Items appearing only in this schema
+        will be prefixed with "-" in the diff.
+    flags : `int`
+        A bitwise OR of :py:class:`lsst.afw.table.Schema.ComparisonFlags`
+        indicating which features of schema items to compare.  The returned
+        diff will always show all differences, but no diff will be shown if
+        the only differences are not included in the flags.  Default is
+        `lsst.afw.table.Schema.IDENTICAL`, which checks everything.
```

Comment:
It seems a bit odd to show diffs that are not in the flags, but I'm guessing that made the code simpler.

In any case it's a minor point. This seems like a really useful utility.